lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1556,20 +1589,28 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
     SPIRVBuiltinVariableKind BVKind;
     if (BVar->isBuiltin(&BVKind))
       BV->setName(prefixSPIRVName(SPIRVBuiltInNameMap::map(BVKind)));
+    bool IsCtorOrDtor = (BV->getName() == "llvm.global_ctors" ||
+                         BV->getName() == "llvm.global_dtors");
+    if (Init && IsCtorOrDtor) {
+      Initializer = dyn_cast<Constant>(transValue(Init, F, BB, false));
+      postProcGlobalCtorDtorTypeInit(Ty, Initializer);
+    }
```

Comment:
It's confusing to have this check separated from if-statement below, and to repeat the check for `IsCtorOrDtor` twice, here and below.